spec/features/merchant_dashboards/index_spec.rb

```diff
@@ -0,0 +1,115 @@
+      @transaction_8 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_9 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_10 = @invoice_5.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'success')
+      @transaction_11 = @invoice_6.transactions.create(credit_card_number: 1234123412341234, credit_card_expiration_date: '2012-03-27', result: 'failed')
+
+      visit "/merchants/#{@merchant.id}/dashboard"
+    end
+
+    it "I see the name of my merchant" do
+      expect(page).to have_content(@merchant.name)
+    end
+
+    it "I see link to merchant items index" do
+      click_link "My Items"
+      expect(current_path).to eq("/merchants/#{@merchant.id}/items")
+    end
+
+    it "I see a link to merchant invoice index" do
+      click_link "My Invoices"
+      expect(current_path).to eq("/merchants/#{@merchant.id}/invoices")
+    end
+
+    it "I see the 5 customer names with the most successful transactions" do
+      visit "/merchants/#{@merchant.id}/dashboard"
+
+      expect(page).to have_content 'Favorite Customers'
+      expect(page).to_not have_content @customer_6.first_name
+      expect(@customer_5.last_name).to appear_before(@customer_1.last_name)
+      expect(@customer_5.last_name).to appear_before(@customer_2.last_name)
+      expect(@customer_5.last_name).to appear_before(@customer_4.last_name)
+      expect(@customer_1.last_name).to appear_before(@customer_3.last_name)
+      expect(@customer_2.last_name).to appear_before(@customer_3.last_name)
+      expect(@customer_4.last_name).to appear_before(@customer_3.last_name)
+    end
+
+    it 'I also see each customers number of succesful transactions' do
+      expect(page).to have_content "#{@customer_5.last_name} - 3 purchases"
+      expect(page).to have_content "#{@customer_1.last_name} - 2 purchases"
+      expect(page).to have_content "#{@customer_2.last_name} - 2 purchases"
+      expect(page).to have_content "#{@customer_4.last_name} - 2 purchases"
+      expect(page).to have_content "#{@customer_3.last_name} - 1 purchases"
+    end
+
+    it 'I see a section for "Items Ready to Ship"' do
+      within ("div#items-ready-to-ship") do
+        expect(page).to have_content("Items Ready to Ship")
+      end
+    end
+
+    it "I see each invoice id as a link to my merchant's invoice show page" do
+      within ("div#items-ready-to-ship") do
+        click_on "#{@invoice_6.id}"
+        expect(current_path).to eq("/merchants/#{@merchant.id}/invoices/#{@invoice_6.id}")
+
+        visit "/merchants/#{@merchant.id}/dashboard"
+
+        click_on "#{@invoice_7.id}"
+        expect(current_path).to eq("/merchants/#{@merchant.id}/invoices/#{@invoice_7.id}")
```

Comment:
I dont know if we would need to test both of the links, if one works the other one should work as well